### PR TITLE
Update blaze to use the Spark DataFrame API

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

**Comment by cpcloud** [comment]
will need to merge the odo PR first https://github.com/ContinuumIO/odo/pull/117

**Comment by cpcloud** [comment]
needs this PR first:

https://github.com/ContinuumIO/odo/pull/139

**Comment by cpcloud** [comment]
@mrocklin any comments?

**Comment by mrocklin** [comment]
I'm sufficiently ignorant here to have no comments.

File: blaze/compute/spark.py [hunk 1]
```diff
@@ -124,8 +128,25 @@ def compute_up(t, rdd, **kwargs):
...
+        return getattr(data, attr, default)
+    except py4j.protocol.Py4JJavaError:
+        return default
+
+
+@compute_up.register(Join, SparkDataFrame, SparkDataFrame)
+def spark_df_join(t, lhs, rhs, **kwargs):
+    # ship to rdd land, so we can reuse handling of combining records code
+    rdd = compute_up(t, lhs.rdd, rhs.rdd, **kwargs)
+    return lhs.sql_ctx.createDataFrame(rdd)
```
Comment (cpcloud) [inline blaze/compute/spark.py]: we might want to eventually do this with `lhs.join(lhs.column == rhs.column, 'inner').selectExprs(*list_of_spark_Column_instances)`, I just didn't want to reimplement the `pair_assemble` logic

**Pull request merged**